engine/src/main/java/org/terasology/network/BroadCastServer.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.network;
+
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.Executors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BroadCastServer {
+
+    private static final Logger logger = LoggerFactory.getLogger(BroadCastServer.class);
+    private static final String DISCOVERY_REQUEST= "DISCOVERY_REQUEST";
+    private static final String DISCOVERY_RESPONSE = "DISCOVERY_RESPONSE";
+
+    private static boolean turnOnBroadcast;
+
+    private DatagramSocket receiveSocket;
+    private DatagramSocket sendSocket;
+
+    public BroadCastServer(){
```

Comment:
You've got some trivial code style issues here and elsewhere, like where we put extra spaces. Please be sure to hook up Checkstyle and have it analyze your newly added code :-)

See https://github.com/MovingBlocks/Terasology/wiki/Checkstyle

Reply:
My bad didn't go through it before will make the changes.